### PR TITLE
[SYSTEMDS-3744, SYSTEMDS-3745, SYSTEMDS-3746, SYSTEMDS-3747, SYSTEMDS-3748] Python API Builtin cumsum, cumprod, summin, cummax, cumsumprod

### PR DESCRIPTION
**Pull request by e-strauss**
[SYSTEMDS-3744, SYSTEMDS-3745, SYSTEMDS-3746, SYSTEMDS-3747, SYSTEMDS-3748] Python API Builtin cumsum, cumprod, summin, cummax, cumsumprod

This patch adds the builtin operators cumsum, cumprod, summin, cummax, cumsumprod to the python api.

### CODE REVIEW

File: src/main/python/systemds/operator/nodes/matrix.py [hunk 1]
```diff
@@ -522,5 +522,12 @@ def replace(self, pattern: VALID_INPUT_TYPES, replacement: VALID_INPUT_TYPES) ->
         """
         return Matrix(self.sds_context, "replace", named_input_nodes={"target": self, "pattern": pattern, "replacement": replacement})
 
+    def cumsum(self) -> 'Matrix':
+        """ Column prefix-sum. (For row-prefix sum, use X.t().cumsum().t())
```
Comment (Baunsgaard) [inline src/main/python/systemds/operator/nodes/matrix.py:526]: interesting, do we not support cumulative sum on rows?

Comment (e-strauss) [reply src/main/python/systemds/operator/nodes/matrix.py:526]: jep, afaik. It says so in the docu (even Arnab told me yesterday not trust it ^^) and java code suggests it too

**Comment by e-strauss** [comment]
reformatted

File: src/main/python/systemds/operator/nodes/matrix.py (hunk 1 above)
Comment (Baunsgaard) [reply src/main/python/systemds/operator/nodes/matrix.py]: @mboehm7 

I think we should probably make a task for this, if it really is not supported.
It seems odd to to two transposes to do row cumulative operations.

Comment (mboehm7) [reply src/main/python/systemds/operator/nodes/matrix.py]: well, its fine to add a task even through the focus was always R compliance and are defines it as column-wise cumulative aggregates (unary operation without additional parameters). The same also applies to rev(). Recently some of the operations like roll() are more inspired by numpy and there you have indeed usually axis parameters.

Comment (mboehm7) [reply src/main/python/systemds/operator/nodes/matrix.py]: Also when committing this task here lets call it [SYSTEMDS-3744-48] ... and in the future let's create JIRA tasks for groups of operations that belong together. Thanks.

Comment (Baunsgaard) [reply src/main/python/systemds/operator/nodes/matrix.py]: i have made a task for it and marked it as a student project, unrelated to this PR, such that we do not forget it.

**Pull request closed without merging**